src/rust/multi.webtoon/src/parser.rs

```diff
@@ -0,0 +1,426 @@
+use aidoku::{
+	error::Result,
+	prelude::format,
+	std::defaults::defaults_get,
+	std::net::Request,
+	std::{String, Vec},
+	Chapter, DeepLink, Filter, Listing, Manga, MangaPageResult, MangaStatus, MangaViewer, Page,
+};
+
+use crate::helper::*;
+
+pub fn parse_manga_list(base_url: String, filters: Vec<Filter>) -> Result<MangaPageResult> {
+	let (query, search) = check_for_search(filters);
+
+	let url = {
+		if search {
+			format!("{}/search?keyword={}", base_url, query)
+		} else {
+			// This is to handle parse_manga_listing as it passes in full a url,
+			// not just the base
+			if base_url.contains("genre") {
+				base_url
+			} else {
+				format!("{}/genre", base_url)
+			}
+		}
+	};
+
+	let html = request(&url, false).html().expect("Failed to get html");
+
+	let mut mangas: Vec<Manga> = Vec::new();
+
+	// Canvas series are series uploaded by individual artists,
+	// aka unlicensed series
+	let canvas_series = defaults_get("canvasSeries")?.as_bool().unwrap_or(true);
+
+	let selector = {
+		if canvas_series {
+			"#content > div.card_wrap ul > li > a"
+		} else {
+			"#content > div.card_wrap ul.card_lst li > a"
+		}
+	};
+
+	for manga in html.select(selector).array() {
+		let manga_node = manga.as_node().expect("Failed to get manga node");
+		let id = get_manga_id(manga_node.attr("href").read());
+		let cover = manga_node.select("img").attr("src").read();
+		let title = manga_node.select(".subj").text().read();
+		let categories = [manga_node.select(".genre").text().read()].to_vec();
+
+		let author_artist = manga_node.select(".author").text().read();
+		let author_artist = author_artist.split('/').collect::<Vec<&str>>();
+
+		let author = String::from(author_artist[0].trim());
+		let mut artist = String::new();
+
+		if author_artist.len() > 1 {
+			artist = String::from(author_artist[1].trim());
+		}
```

Comment:
I don't think it's worth it to parse any of these -- the app doesn't use or expect them, and the extra scraping of info slows it down a bit. Best to leave the details parsing for the manga details function.